state/unit.go

```diff
@@ -932,8 +935,49 @@ func (u *Unit) PrivateAddress() (network.Address, error) {
 	return m.PrivateAddress()
 }
 
-// containerAddress returns the address of the container (pod)
-// in which the unit workload is running.
+func (u *Unit) scopedAddress(scope string) (network.Address, error) {
+	addr, err := u.serviceAddress(scope)
+	if err == nil {
+		return addr, nil
+	}
+	if network.IsNoAddressError(err) {
+		return u.containerAddress()
+	}
+	return network.Address{}, errors.Trace(err)
+}
+
+// AllAddresses returns the public and private addresses
+// plus the container address of the unit (if known).
+// Only relevant for CAAS models - will return an empty
```

Comment:
Because of this comment, I think we can short-circuit all this logic for unit on IAAS models?

Reply:
I would have liked to see a test for a unit.AllAddresses() in IAAS model too.

Reply:
added test and short circuit